frontends/web/src/routes/buy/utils.tsx

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExchangeDeals } from '../../api/exchanges';
+import { ExchangeProviderWithBestDeal, PaymentMethod } from './types';
+
+function findLowestFee(exchangeDeals: ExchangeDeals) {
+  let lowestFee = Object.values(exchangeDeals).map(providerDeals => Math.min(...providerDeals.map(deal => deal.fee)));
+  return Math.min(...lowestFee);
+}
+
+export function findBestDeal(exchangeDeals: ExchangeDeals): ExchangeProviderWithBestDeal[] {
+  // Get the lowest fee from all providers
+  const lowestFee = findLowestFee(exchangeDeals);
+  // Loop through each provider in the exchangeDeals object
+  let arr: ExchangeProviderWithBestDeal[] = [];
+  let provider: keyof ExchangeDeals;
+  for (provider in exchangeDeals) {
+    const providerObj = { providerName: provider, methods: exchangeDeals[provider] as PaymentMethod[] };
+    arr = [...arr, providerObj];
+  }
+  //returns array of object of all providers
+  //and each of its payment method has `isBestDeal` property
+  return arr.map(provider => ({
+    ...provider,
+    methods: provider.methods.map(method => ({
+      ...method,
+      isBestDeal: method.fee === lowestFee
+    })).sort((a, b) => a.payment.localeCompare(b.payment))
+  }));
+}
```

Comment:
These utils are created to change the data structure of the exchangeDeal (from BE). Now, the data that we use look like this:

The data is now restructured into this: `ExchangeProviderWithBestDeal` :


```js

type ExchangeProviderWithBestDeal = {
    providerName: string;
    methods: PaymentMethod[];
    supported?: boolean;
}

type PaymentMethod = ExchangeDeal & {
    isBestDeal: boolean;
}

//this ExchangeDeal already exists before
type ExchangeDeal = {
  fee: number;
  payment: 'card' | 'bank-transfer';
  isFast: boolean;
}
```

This way, it's now really easy to render all exchanges and its deals, including everything such as the "bestDeal" inside the object itself.